src/rpc.rs

```diff
@@ -431,7 +441,7 @@ impl<T: Runtime> Rpc<T> {
         let ext_hash = T::Hashing::hash_of(&extrinsic);
         log::info!("Submitting Extrinsic `{:?}`", ext_hash);
 
-        let events_sub = self.subscribe_events().await?;
+        let events_sub = self.subscribe_finalized_events().await?;
```

Comment:
We might want to make this configurable, but I saw no reason not to wait for finalized events here.